CMakeLists.txt

```
@@ -17,22 +18,45 @@ option(DOWNLOAD_ENSMALLEN "If ensmallen is not found, download it." ON)
 option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
+set(BOARD_NAME "" CACHE INTERNAL "Enter the board name to compile mlpack for.")
+
+if (CMAKE_CROSSCOMPILING)
+  include(board/flags-config.cmake)
+  if (BOARD_NAME STRGREATER  "")
```

```suggestion
  if (BOARD_NAME)
```

I think, but I'm not 100% sure of this.

---

Neither do I, but I do remember as this is the only way to check if a string is empty, let me check on this locally before commiting